projects/canopy/src/lib/forms/input/docs/guide.stories.mdx

```diff
@@ -0,0 +1,158 @@
+</lg-input-field>
+```
+
+## How it works
+
+A text input should always use the `<label>` and `<input>` contained in the component.
+
+The text input consists of an interactive container which upon selection becomes active allowing the user to enter text via the keyboard or other input device. Labelling above the text input gives context as to the information required and all inputs must include a label.
+
+The user can navigate between inputs using the tab key on desktop, and forward / back icons on mobile (OS dependent). Once finished, the user can exit the input by clicking outside of the container on desktop, or by tapping the done button on mobile (OS dependent).
+
+### Prefixes and suffixes
+
+You can add prefixes (i.e. currency symbols), suffixes (i.e. percentage) using the `lgSuffix` and `lgPrefix` directives.
+
+![prefix-suffix](docs/text-input/prefix-suffix.png)
+
+#### Text prefix
+
+```html
+<lg-input-field>
+  Name
+  <input lgInput formControlName="name" />
+  <span lgPrefix>£</span>
+</lg-input-field>
+```
+
+#### Text suffix
+
+```html
+<lg-input-field>
+  Name
+  <input lgInput formControlName="name" />
+  <span lgSuffix>%</span>
+</lg-input-field>
+```
+
+### Input buttons
+
+Buttons can be added within input fields to provide functionality linked to the input. The `lgSuffix` directive needs to be added to the button to place it in the correct place. Only small size buttons should be placed in the input field.
+
+![button](docs/text-input/button.png)
+
+#### Button suffix
+
+```html
+<lg-input-field>
+  Name
+  <input lgInput formControlName="name" />
+  <button lg-button lgSuffix size="sm">
+    Search
+  </button>
+</lg-input-field>
+```
+
+#### Icon Button suffix with add-on class
+
+There is an additional add-on button variant for adding a button to the input field which inherits the button spacing but no background or border colours.
+
+```html
+<lg-input-field>
+  Name
+  <input lgInput formControlName="name" />
+  <button lg-button lgSuffix size="sm" [iconButton]="true" variant="add-on">
+    <lg-icon name="search" />
+  </button>
+</lg-input-field>
+```
+
+## Dos and Don'ts
+
+### Do
+
+![do](docs/text-input/do.png)
+
+1. **Do** use labels and supporting text to help provide context
+2. **Do** keep labels clear, concise and relevant
+3. **Do** use a textarea for multi-line content (Angular component to be built)
+4. **Do** use prefix, suffix or action variants to aid the user as to the input type or information required
+5. **Do** keep input widths relative to the information requested
+
+### Don't
+
+![dont](docs/text-input/dont.png)
+
+1. **Don't** truncate label text
+2. **Don't** use text inputs when a pre-defined response is required, use the select component instead
+3. **Don't** use text inputs for simple responses (such as yes/no), use checkboxes or radio buttons instead
+4. **Don't** use placeholder text instead of a label
+
+## Additional development information
+
```

Comment:
Could we add this line here please? **We recommend using <a href="https://angular.io/guide/reactive-forms" target="_blank">Angular's reactive forms</a>.**